### PR TITLE
Update dependencies to work with Vite and NextJS, combine css files into a single css file

### PR DESCRIPTION
**Pull request by vladimir-mck**
## Description
Resolves  [#1293](https://github.com/kedro-org/kedro-viz/issues/1293), [#1504](https://github.com/kedro-org/kedro-viz/issues/1504) and [1512](https://github.com/kedro-org/kedro-viz/issues/1512)

## Development notes

This pull request addresses several JavaScript issues, enhancing the compatibility of Kedro-Viz when integrated with Svelte and Next.js. Although the JS-related errors have been resolved.

- Replaced lodash. dependencies with lodash to ensure compatibility with both Next.js and Vite.
- Upgraded Plotly to a newer version to fix errors experienced in both Vite and Next.js environments.
- Removed css imports from components and replaced with scsc imports
- Remove script to transform scss into css in dev mode
- Added script to combine lib css files into a single css file 

Vite: 
Screenshot 2023-08-29 at 13 17 24
Screenshot 2023-08-29 at 13 17 34

## QA notes

Create a new project with Vite:
- npm create vite@latest my-react-app --template react
- install kedro-viz alpha version `npm i @quantumblack/kedro-viz@6.5.1-alpha.3`
- add the component and styles
- data to it.

Create a new project with NextJS:
- npx create-next-app@latest
- install kedro-viz alpha version `npm i @quantumblack/kedro-viz@6.5.1-alpha.3`
- Import kedro following update readme

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1508"></a>

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1510"></a>

### CODE REVIEW

**Review by ravi-kumar-pilla — APPROVED** [review]
Nice work Vlad !!

Some questions -
1. I want to understand why we need apollo config refactoring ?   
2. I have tested the PR using the QA notes,  the alpha release works well except for the CSS errors (as you mentioned). Just to confirm, I am attaching the errors during the run

Svelte error -

Screenshot 2023-08-31 at 5 54 08 PM

Next.js error -

Screenshot 2023-08-31 at 6 02 31 PM

**Comment by vladimir-mck** [comment]
Hey Ravi
> 1. I want to understand why we need apollo config refactoring ?

So here we needed a bit of a refactor because it tries to access the `window.location` object to get the host, path name and protocol. The window object doesn't exist when it's rendered on the server, additionally the web socket link will also throw an error. 

> 2. I have tested the PR using the QA notes,  the alpha release works well except for the CSS errors (as you mentioned). Just to confirm, I am attaching the errors during the run

Yep, that's okay. The css error is not present when started with Vite. Next has it's own rules how to handle css and that's why it's throwing an error. I have to agree with them on this, we shouldn't release with 66 css files, one is enough.

File: .github/workflows/check-release.yml [hunk 1]
```diff
@@ -8,7 +8,6 @@ on:
 jobs:
   check-version:
     runs-on: ubuntu-latest
-    if: startsWith(github.head_ref, 'release/') || startsWith(github.ref, 'refs/heads/release/')
```
Comment (vladimir-mck) [inline .github/workflows/check-release.yml:11]: This and all the rest of the python changes are already on main. I checked main and I don't know why it's showing it as new code. Any idea ?

**Review by tynandebold — COMMENTED** [review]
The code is looking fantastic. I'm excited for this change to go in.

One thing when I'm trying to test is I can't install the alpha version of the package. I get this error: 

image

Do you see the same?

**Review by tynandebold — COMMENTED** [review]
I tried this out using create react app and everything loads and works fine. I see that some of the styles are off though. Have a look:

Screenshot 2023-09-08 at 11 18 06



This is the alpha version I'm using: `"@quantumblack/kedro-viz": "^6.5.1-alpha.2"`.

File: RELEASE.md [hunk 2]
```diff
@@ -17,7 +23,7 @@ Please follow the established format:
 - Fix to search for a '<lambda' Python function in the sidebar. (#1497)
 - Add favicon to Kedro-Viz. (#1509)
 - Remove python upper-bound requirements and add KedroVizPythonVersion warning. (#1506)
-
+- Resolved various bugs affecting server-side rendering, updated dependencies to ensure compatibility with Vite and Next.js environments (#1508)
```
Comment (tynandebold) [inline RELEASE.md:26]: Why is this here? That [PR](https://github.com/kedro-org/kedro-viz/pull/1508) was never merged, right?

File: RELEASE.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ Please follow the established format:
 - Include the ID number for the related PR (or PRs) in parentheses
 -->
 
+# Release 6.5.1
+
+## Bug fixes and other changes
+
+- Updated dependencies to ensure compatibility with Vite and Next.js environments, combine css into a signle file for npm library (#1510)
```
Comment (tynandebold) [inline RELEASE.md:13]:
```suggestion
- Updated dependencies to ensure compatibility with Vite and Next.js environments; combine CSS into a single file when used as a React component. (#1510)
```

**Review by tynandebold — APPROVED** [review]
Really nice! Everything is looking great and working well.

I've left a couple small comments, nothing blocking.

File: RELEASE.md (hunk 2 above)
Comment (ravi-kumar-pilla) [reply RELEASE.md:31]: +1

File: README.md [hunk 4]
```diff
@@ -154,16 +154,26 @@ train_evaluation.r2_score_linear_regression:
...
+
+To use with NextJS:
+
+```javascript
+import '@quantumblack/kedro-viz/lib/styles/styles.min.css';
+import dynamic from 'next/dynamic';
+
+const NoSSRKedro = dynamic(() => import('@quantumblack/kedro-viz'), {
+  ssr: false,
+});
```
Comment (ravi-kumar-pilla) [inline README.md:174]: Awesome !!

**Review by ravi-kumar-pilla — APPROVED** [review]
Awesome work Vlad 👍 . Need some changes with the release doc but nothing blocking.
